File: easy/1018_binary_prefix_divisible_by_5/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<bool> prefixesDivBy5(std::vector<int> &nums) {
    int val = 0;
    std::vector<bool> answer;

    for (int i = 0; i < nums.size(); i++) {
      val = (val * 2 + nums[i]) % 5;
      answer.push_back(val == 0);
    }

    return answer;
  }
};

File: easy/1018_binary_prefix_divisible_by_5/solution.py
from typing import List


class Solution:
    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        bin_num = 0
        answer = []

        for i in range(len(nums)):
            bin_num = (bin_num << 1) | nums[i]
            answer.append(bin_num % 5 == 0)

        return answer
